### ISSUE TITLE
Crasher at Realm() init regarding Object subclass property(s)

### ISSUE THREAD

**Issue by doonware** [realm/realm-cocoa#6406]
## Goals
Upgraded from v.3.19.0 -> 4.3.0, need to resolve crasher when opening/initializing a Realm().

## Expected Results
Open Realm w/o issue, or if there is a compatibility issue, provide context in debug console as to which subclass or property is of concern.

This is not a migration issue as it occurs on a brand new app install. Also not a startup race condition as it requires the user to login and authenticate with our servers prior to opening a Realm.

## Actual Results
When attempting to initialize a new Realm (fresh install) an exception is thrown during `ObjectUtil.getSwiftProperties()`.

It does not matter if it is the _default.realm_ from base `Realm()` initializer or via `Realm(configuration:)` for our encrypted user realm. Any time a Realm is opened and goes through the process of inspecting the Object subclasses' properties for schema related matters the lib crashes.

I have tried setting `dead code snipping` to `no` as I have seen similar issues from other devs in relation to that. However that did not influence this in any way.

## Steps to Reproduce
For me it occurs simply by calling `try Realm()` or `try Realm(configuration:)`. Without providing all our Realm data models (which I cannot really do) I am not sure how to attempt repro.

I am pretty confident it has something to do with one (perhaps more) of our Object subclasses but there is just 0 context as to what object(s) that may be, the properties involved in the issue, etc. We have 30 or so different data  models. Everything was working on the previous version of Realm utilized (3.19.0).

![image](https://user-images.githubusercontent.com/3248753/72573505-c6299180-387a-11ea-88f4-2e0d866b99cd.png)

## Version of Realm and Tooling

Realm framework version: 4.3.0

Realm Object Server version: None

Xcode version: 11.3.1

iOS/OSX version: 13.3

Dependency manager + version: Carthage 0.34.0

**Comment by tgoyne** [comment]
The stack trace shows an exception being thrown. What is that exception's message?

**Comment by doonware** [comment]
I don't get any message displayed in debugger unfortunately. If I let execution continue beyond the exception breakpoint then it results in a `SIGABRT` and then `EXC_BREAKPOINT (code=1, subcode=0x19c47ab14)` at AppDelegate

**Comment by doonware** [comment]
Ok I figured it out. The issue had to do with using `@objc enum` as a property in the managed object. Apparently there is now a `RealmEnum` protocol which these enums need to conform too.

I didn't see any mention of this in the release notes so it took a lot of trial and error editing our data models in order to track it down. A nice resolution for this ticket would be to add this new protocol requirement in the release notes in the event other users encounter it. It would also be great to get some debug messaging that would explicitly call this out.

Closing the ticket as my issue is now resolved.
